src/dbnode/persist/fs/retriever.go

```diff
@@ -288,6 +288,13 @@ func (r *blockRetriever) fetchBatch(
 	reqs []*retrieveRequest,
 	seekerResources ReusableSeekerResources,
 ) {
+	if err := r.opts.QueryStats().UpdateBytesRead(0); err != nil {
```

Comment:
a comment would help here. not obvious you are checking to see if the limit has already been exhausted

Reply:
I think better yet if we made this into a standalone function? i.e. `r.opts.QueryStats().Exhausted()` which would return a bool?